spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java

```diff
@@ -233,10 +241,18 @@ private String getBrokerToken(AuthenticationMethod credentialType, String authHe
 					logger.debug("return (client-credentials) access token for {} from cache", cacheKey);
 					return cachedToken;
 				} else {
-					String token = tokenBroker
-							.getAccessTokenFromClientCredentials(oauthTokenUrl, clientCredentialsFromHeader.getId(),
-									clientCredentialsFromHeader.getSecret());
-					tokenCache.put(cacheKey, token);
+					String token;
+					if (tokenBroker == null) {
+						token = oAuth2TokenService.retrieveAccessTokenViaClientCredentialsGrant(
+								URI.create(oauthTokenUrl), clientIdentity, null, null,
+								null, false).getAccessToken();
+					} else {
+						// only when deprecated constructors are used, for backward compatibility
+						token = tokenBroker
+								.getAccessTokenFromClientCredentials(oauthTokenUrl, clientCredentialsFromHeader.getId(),
+										clientCredentialsFromHeader.getSecret());
+						tokenCache.put(cacheKey, token);
```

Comment:
tokenCache must be filled outside the if/else clause when token is not null

Reply:
good point 👍🏻